devlib/trace/ftrace.py

```diff
@@ -225,6 +225,8 @@ def reset(self):
             self._set_buffer_size()
         self.target.execute('{} reset'.format(self.target_binary),
                             as_root=True, timeout=TIMEOUT)
+        self.target.execute('echo 0 > {}'.format(self.function_profile_file),
```

Comment:
This currently will be executed regardless of whether the function profiler was to be used / supported on the device which can cause errors if unavailable. 
I would suggest adding a guard here as in other parts of the code as to whether this should be reset.